pkg/controllers/user/pipeline/engine/jenkins/engine.go

```diff
@@ -329,10 +329,16 @@ func (j *Engine) SyncExecution(execution *v3.PipelineExecution) (bool, error) {
 		updated = true
 		execution.Labels[utils.PipelineFinishLabel] = "true"
 		execution.Status.ExecutionState = utils.StateSuccess
+		if v3.PipelineExecutionConditonProvisioned.IsUnknown(execution) {
+			v3.PipelineExecutionConditonProvisioned.True(execution)
+		}
 	} else if info.Status == "FAILED" && execution.Status.ExecutionState != utils.StateFail {
 		updated = true
 		execution.Labels[utils.PipelineFinishLabel] = "true"
 		execution.Status.ExecutionState = utils.StateFail
+		if v3.PipelineExecutionConditonProvisioned.IsUnknown(execution) {
+			v3.PipelineExecutionConditonProvisioned.True(execution)
```

Comment:
If the status is failed, shouldn't we set the PipelineExecutionConditonProvisioned to false?

Reply:
In Rancher we try to follow this practice when it comes to condition management:

* True means that things pass. What is the reason why we need this condition if it gets set to True in all the cases when execution just completed, no matter whether it's true or false? https://github.com/gitlawr/rancher/blob/c45029201df98ffc842a94075271e3d34594a95c/pkg/controllers/user/pipeline/engine/jenkins/engine.go#L333 and https://github.com/gitlawr/rancher/blob/c45029201df98ffc842a94075271e3d34594a95c/pkg/controllers/user/pipeline/engine/jenkins/engine.go#L340
Why do we need another condition to record execution result? why can't we use the same one? Assuming that if one of the condition fails, the logic would rerun again.
Or is the main point of `PipelineExecutionConditonProvisioned` is to indicate when the stuff is in provision-ing (Uknown) state, and we evaluate it somewhere else in the code?

* We should take advantage of condition framework when it comes to condition update. But I guess this won't work if both failure and success are getting resulted in condition set to true

Reply:
The inial idea is to differentiate unexpected failures like connection issues and expected failures like scripts exit 1. I think you are right, we don't need 2 conditions for that. If it fails unexpectedly mark the condition as false otherwise mark it as true. Does it make sense to you?

Reply:
yes, that would be ideal. For that, you can also use the norman condition framework where your func is passed as a callback